Shopify/tophat
# Update `tophatctl` so that it can wait for a response from Tophat

### What does this change accomplish?

This change accomplishes two things:
- Updates the underlying distributed notification code so that it can wait for a reply to a message.
- Updates `tophatctl install` to wait for the install to be completed before exiting.

### How have you achieved it?

- By abstracting await the concept of notifications (`TophatInterProcessNotifier`) into a new set of utilities called `TophatControlServices`.
- By adding support for `Reply` types on sent requests and adding the appropriate code to emit the replies.
- By converting `NotificationHandler` to `RemoteControlReceiver` and having it handle replies accordingly.
- By updating `tophatctl install` to wait for the reply at the end.

### How can the change be tested?

Run `tophatctl install <arguments>` and ensure that the command waits for install to complete (or fail).
